src/Microsoft.ML.Core/Data/ReadOnlyMemoryUtils.cs

```diff
@@ -0,0 +1,397 @@
+                        right = default;
+                        return false;
+                    }
+                    // REVIEW: Can this be faster?
+                    if (ContainsChar(span[ichCur], separators))
+                        break;
+                }
+            }
+
+            left = memory.Slice(0, ichCur);
+            right = memory.Slice(ichCur + 1, memory.Length - ichCur - 1);
+            return true;
+        }
+
+        /// <summary>
+        /// Returns a <see cref="ReadOnlyMemory{T}"/> of <see cref="char"/> with leading and trailing spaces trimmed. Note that this
+        /// will remove only spaces, not any form of whitespace.
+        /// </summary>
+        public static ReadOnlyMemory<char> Trim(ReadOnlyMemory<char> memory)
+        {
+            if (memory.IsEmpty)
+                return memory;
+
+            int ichLim = memory.Length;
+            int ichMin = 0;
+            var span = memory.Span;
+            if (span[ichMin] != ' ' && span[ichLim - 1] != ' ')
+                return memory;
+
+            while (ichMin < ichLim && span[ichMin] == ' ')
+                ichMin++;
+            while (ichMin < ichLim && span[ichLim - 1] == ' ')
+                ichLim--;
+            return memory.Slice(ichMin, ichLim - ichMin);
+        }
+
+        /// <summary>
+        /// Returns a <see cref="ReadOnlyMemory{T}"/> of <see cref="char"/> with leading and trailing whitespace trimmed.
+        /// </summary>
+        public static ReadOnlyMemory<char> TrimWhiteSpace(ReadOnlyMemory<char> memory)
+        {
+            if (memory.IsEmpty)
+                return memory;
+
+            int ichMin = 0;
+            int ichLim = memory.Length;
+            var span = memory.Span;
+            if (!char.IsWhiteSpace(span[ichMin]) && !char.IsWhiteSpace(span[ichLim - 1]))
+                return memory;
+
+            while (ichMin < ichLim && char.IsWhiteSpace(span[ichMin]))
+                ichMin++;
+            while (ichMin < ichLim && char.IsWhiteSpace(span[ichLim - 1]))
+                ichLim--;
+
+            return memory.Slice(ichMin, ichLim - ichMin);
+        }
+
+        /// <summary>
+        /// Returns a <see cref="ReadOnlyMemory{T}"/> of <see cref="char"/> with trailing whitespace trimmed.
+        /// </summary>
+        public static ReadOnlyMemory<char> TrimEndWhiteSpace(ReadOnlyMemory<char> memory)
+        {
+            if (memory.IsEmpty)
+                return memory;
+
+            int ichLim = memory.Length;
+            var span = memory.Span;
+            if (!char.IsWhiteSpace(span[ichLim - 1]))
+                return memory;
+
+            while (0 < ichLim && char.IsWhiteSpace(span[ichLim - 1]))
+                ichLim--;
+
+            return memory.Slice(0, ichLim);
+        }
+
+        /// <summary>
+        /// This produces zero for an empty string.
+        /// </summary>
+        public static bool TryParse(out Single value, ReadOnlyMemory<char> memory)
```

Comment:
This feels wrong to have `Parse` on `DoubleParser` and `TryParse` on ReadOnlyMemoryUtils.  I think they should go together.

Reply:
Yes. I shouldn't have to go to `DoubleParser` to call `Parse` and then `ReadOnlyMemoryUtils` to call `TryParse`. The methods go together and belong in the same place.